libraries/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts

```diff
@@ -459,6 +465,39 @@ export class PhasedScriptAction extends BaseScriptAction<IPhasedCommandConfig> {
     await this._executeOperations(initialOptions);
   }
 
+  private _registerWatchModeInterface(): void {
+    const terminal: ITerminal = this._terminal;
+
+    process.stdin.setRawMode(true);
+    process.stdin.resume();
+    process.stdin.setEncoding('utf8');
+    process.stdin.on('data', (key: string) => {
+      if (key === this._toggleWatcherKey) {
+        this._isWatcherPaused = !this._isWatcherPaused;
+        if (this._isWatcherPaused) {
+          terminal.writeLine(
+            `Project watcher will pause after the build. Press '${this._toggleWatcherKey}' to resume.`
+          );
+        } else {
+          terminal.writeLine(`Project watcher resumed. Press '${this._toggleWatcherKey}' to pause.`);
+        }
+      }
+      if (key === this._buildOnceKey) {
+        if (this._isWatcherPaused) {
+          terminal.writeLine(`Building once...`);
+          this._isWatcherPaused = false;
+          setTimeout(() => {
+            this._isWatcherPaused = true;
+          }, 1000);
```

Comment:
This should just be able to call `projectWatcher.resume(); projectWatcher.pause();` and internally that should result in an immediate invocation of `projectWatcher`'s internal `resolveIfChanged` callback, and toggling a flag back and forth that controls whether or not file events invoke the callback.